File: frontend/src/types/station.ts
export interface Station {
  fmisid: number
  name: string
  region: string | null
  country: string
  latitude: number
  longitude: number
  elevation: number | null
  station_type: string | null
  is_active: boolean
}

export interface StationLatest {
  fmisid: number
  name: string
  region: string | null
  latitude: number
  longitude: number
  time: string | null
  temperature: number | null
  humidity: number | null
  wind_speed: number | null
  wind_direction: number | null
  wind_gust: number | null
  precipitation_1h: number | null
  snow_depth: number | null
  pressure: number | null
}

File: backend/app/schemas/station.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class StationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    fmisid: int
    name: str
    region: Optional[str] = None
    country: Optional[str] = "Finland"
    latitude: float
    longitude: float
    elevation: Optional[float] = None
    station_type: Optional[str] = None
    is_active: bool = True


class StationLatest(BaseModel):
    fmisid: int
    name: str
    region: Optional[str] = None
    latitude: float
    longitude: float
    time: Optional[datetime] = None
    temperature: Optional[float] = None
    humidity: Optional[float] = None
    wind_speed: Optional[float] = None
    wind_direction: Optional[int] = None
    wind_gust: Optional[float] = None
    precipitation_1h: Optional[float] = None
    snow_depth: Optional[float] = None
    pressure: Optional[float] = None
